web/views.py

```
@@ -5101,6 +5102,114 @@
+
+            # Append icon_url to the achievement
+            achievement.icon = icon_url
+            achievement.save()
+
+            # Pass achievement ID, wallet address, and icon URL to NFT minting service
+            nft_badge = mint_nft_badge(achievement_id, wallet_address)
+
+            if nft_badge:
+                messages.success(request, f"NFT badge successfully minted and sent to {wallet_address}!")
+                return redirect("achievement_detail", achievement_id=achievement_id)
+            else:
```

Ruff (0.8.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)